api/include/opentelemetry/baggage/baggage_impl.h

```diff
@@ -0,0 +1,292 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cctype>
+
+#include "opentelemetry/common/kv_properties.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+
+namespace baggage
+{
+
+static constexpr size_t kMaxKeyValuePairs = 180;
+static constexpr size_t kMaxKeyValueSize  = 4096;
+static constexpr size_t kMaxSize          = 8192;
+static constexpr char kKeyValueSeparator  = '=';
+static constexpr char kMembersSeparator   = ',';
+static constexpr char kMetadataSeparator  = ';';
+
+class BaggageImpl
```

Comment:
Can this be hidden? Maybe have an "internal" as part of the namespace and we add a rule that everything in the internal package is private? "baggage::internal::BaggageImpl"?